src/media/js/elements/header.js

```diff
@@ -0,0 +1,162 @@
+                    }
+                }
+            }
+        })
+    });
+
+    var MktHeaderChildElement = document.registerElement('mkt-header-child', {
+        prototype: Object.create(HTMLElement.prototype, {
+            createdCallback: {
+                value: function() {
+                    var root = this;
+
+                    if (root.hasAttribute(CHILD_INPUT)) {
+                        // If main element is input, then hide child on blur.
+                        var input = root.querySelector('input');
+                        input.onblur = function() {
+                            setTimeout(function() {
+                                // setTimeout to queue behind toggle button.
+                                root.toggle(false);
+                            }, 50);
+                        };
+                        root.querySelector('form').onsubmit = function() {
+                            setTimeout(function() {
+                                input.value = '';
+                            }, 50);
+                        };
+                    }
+                }
+            },
+            toggle: {
+                value: function(bool) {
+                    // Toggle visibility.
+                    var root = this;
+                    if (bool !== undefined) {
+                        if (bool) {
+                            root.classList.add(CHILD_VISIBLE);
+                        } else {
+                            root.classList.remove(CHILD_VISIBLE);
+                        }
+                    } else {
+                        root.classList.toggle(CHILD_VISIBLE);
+                    }
+
+                    // Keep state on the toggle button. The child is source of
+                    // truth so keep this logic in the child.
+                    var toggleButton = document.querySelector(
+                        'mkt-header-child-toggle[for="' + root.id + '"]');
+                    if (toggleButton) {
+                        toggleButton.classList.toggle(
+                            CHILD_VISIBLE,
+                            root.classList.contains(CHILD_VISIBLE));
+                    }
+
+                    return root;
+                }
+            }
+        })
+    });
+
+    var MktHeaderChildToggleElement = document.registerElement('mkt-header-child-toggle', {
+        prototype: Object.create(HTMLElement.prototype, {
+            createdCallback: {
+                value: function() {
+                    var root = this;
+
+                    root.addEventListener('click', function() {
+                        // On click, find headerChild to toggle.
+                        var headerChild = document.getElementById(
+                            root.getAttribute('for'));
+
+                        if (headerChild.hasAttribute(CHILD_INPUT)) {
+                            // Extra toggle logic since input hides on blur.
```

Comment:
```onblur``` will close the child, but ```<mkt-header-child-toggle>``` toggles it. If you click the toggle button, both the onblur and toggle will fire causing the child to go back and forth. And yeah.